### PR TITLE
Swift conversion review: davidme

### PR DESCRIPTION
**Pull request by davidme-stripe**
These are Swift conversion files for davidme to review. None of them were converted by davidme.

### CODE REVIEW

File: Stripe/STPCard.swift [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+    card.address = STPAddress()
+
+    card.stripeID = stripeId
+    card.name = dict.stp_string(forKey: "name")
+    card.last4 = last4
+    card.dynamicLast4 = dict.stp_string(forKey: "dynamic_last4")
+    card.brand = self.brand(from: rawBrand)
+    //#pragma clang diagnostic push
+    //#pragma clang diagnostic ignored "-Wdeprecated"
+    // This is only intended to be deprecated publicly.
```
Comment (davidme-stripe) [inline Stripe/STPCard.swift:269]: - [ ] @yuki-stripe `funding` got un-deprecated in the conversion: Should we just remove it? It seems like it's been deprecated for a while.

File: Stripe/STPCardValidationState.swift [hunk 2]
```diff
@@ -0,0 +1,9 @@
+//
```
Comment (davidme-stripe) [inline Stripe/STPCardValidationState.swift:1]: - [ ] We should delete this file

File: Stripe/STPCustomer.swift [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+  ) {
+    self.stripeID = stripeID
+    self.defaultSource = defaultSource
+    self.sources = sources
+    self.shippingAddress = shippingAddress
+    self.allResponseFields = allResponseFields
+    super.init()
+  }
+
+  // #warning("Remove when STPAPIRequest doesn't need an instance of this for deserialization")
```
Comment (davidme-stripe) [inline Stripe/STPCustomer.swift:65]: Remove this

File: Stripe/STPMandateOnlineParams.swift [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+extension STPMandateOnlineParams: STPFormEncodable {
+
+  @objc internal var ipAddressField: String? {
+    guard inferFromClient == nil || !(inferFromClient?.boolValue ?? false) else {
+      return nil
+    }
+    return ipAddress
+  }
+
+  @objc internal var userAgentField: String? {
```
Comment (davidme-stripe) [inline Stripe/STPMandateOnlineParams.swift:41]: @csabol-stripe This `inferFromClient` check seems to be new behavior: That's intentional, right?

File: Stripe/STPPaymentMethod.swift [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+      .oxxo, .sofort, .grabPay,  // fall through
+      .unknown:
+      return false
+    @unknown default:
+      return false
+    }
+  }
+}
+
+//
```
Comment (davidme-stripe) [inline Stripe/STPPaymentMethod.swift:306]: Let's remove this

File: Stripe/STPPaymentMethodCardChecks.swift [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+/// Checks on Card address and CVC.
+/// - seealso: https://stripe.com/docs/api/payment_methods/object#payment_method_object-card-checks
+public class STPPaymentMethodCardChecks: NSObject, STPAPIResponseDecodable {
+  override required init() {
+    super.init()
+  }
+
+  // MARK: - Deprecated
+  // TODO(swift): Figure out deprecation strategy
```
Comment (davidme-stripe) [inline Stripe/STPPaymentMethodCardChecks.swift:34]: We should figure out deprecation here

File: Stripe/STPSource.swift [hunk 7]
```diff
@@ -0,0 +1,387 @@
...
+    }
+  }
+
+  @objc public var isReusable: Bool {
+    return usage != .singleUse
+  }
+}
+
+//
+//  STPSource+Private.h
```
Comment (davidme-stripe) [inline Stripe/STPSource.swift:382]: Remove this

File: Stripe/STPSourceCardDetails.swift [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+
+import Foundation
+
+/// The status of this card's 3D Secure support.
+/// - seealso: https://stripe.com/docs/sources/three-d-secure#check-requirement
+@objc
+public enum STPSourceCard3DSecureStatus: Int {
+  /// 3D Secure is required. This card must be converted into a 3D Secure
+  /// source for a charge on it to be successful.
+  case `required`
```
Comment (davidme-stripe) [inline Stripe/STPSourceCardDetails.swift:17]: We should rename these to something less unwieldy

File: Stripe/STPSourceCardDetails.swift [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+  }
+
+  required init(dictionary dict: [AnyHashable: Any]) {
+    allResponseFields = dict
+    let dict = (dict as NSDictionary).stp_dictionaryByRemovingNulls() as NSDictionary
+    last4 = dict.stp_string(forKey: "last4")
+    brand = STPCard.brand(from: dict.stp_string(forKey: "brand") ?? "")
+    //#pragma clang diagnostic push
+    //#pragma clang diagnostic ignored "-Wdeprecated"
+    // This is only intended to be deprecated publicly.
```
Comment (davidme-stripe) [inline Stripe/STPSourceCardDetails.swift:68]: This doesn't seem to be deprecated in ObjC, should we just remove this?

File: Stripe/STPCard.swift (hunk 1 above)
Comment (davidme-stripe) [reply Stripe/STPCard.swift:269]: Actually, it looks like it isn't deprecated in the Objective-C headers.

File: Stripe/STPPaymentHandler.swift [hunk 10]
```diff
@@ -0,0 +1,1562 @@
+//
+//  STPPaymentHandler.swift
+//  StripeiOS
+//
+//  Created by Cameron Sabol on 5/10/19.
+//  Copyright © 2019 Stripe, Inc. All rights reserved.
+//
+
+//
```
Comment (davidme-stripe) [inline Stripe/STPPaymentHandler.swift:9]: Fix this double-comment situation

File: Stripe/STPPaymentHandler.swift [hunk 11]
```diff
@@ -0,0 +1,1562 @@
...
+          if let redirectURL = useStripeSDK.redirectURL {
+            let returnURL: URL?
+            if let returnURLString = currentAction.returnURLString {
+              returnURL = URL(string: returnURLString)
+            } else {
+              returnURL = nil
+            }
+            _handleRedirect(to: redirectURL, withReturn: returnURL)
+          } else {
+            // TOOD : Error
```
Comment (davidme-stripe) [inline Stripe/STPPaymentHandler.swift:960]: We should handle this error case

**Review by davidme-stripe — COMMENTED** [review]
Done with the first pass

File: Stripe/STPPaymentHandler.swift (hunk 11 above)
Comment (davidme-stripe) [reply Stripe/STPPaymentHandler.swift:960]: @csabol

Comment (davidme-stripe) [reply Stripe/STPPaymentHandler.swift:960]: @csabol-stripe

File: Stripe/STPPaymentHandler.swift (hunk 10 above)
Comment (davidme-stripe) [reply Stripe/STPPaymentHandler.swift:9]: @csabol-stripe

File: Stripe/STPPaymentMethodCardChecks.swift (hunk 6 above)
Comment (davidme-stripe) [reply Stripe/STPPaymentMethodCardChecks.swift:34]: @yuki-stripe

File: Stripe/STPPaymentMethod.swift (hunk 5 above)
Comment (davidme-stripe) [reply Stripe/STPPaymentMethod.swift:306]: @yuki-stripe

File: Stripe/STPSourceCardDetails.swift (hunk 8 above)
Comment (davidme-stripe) [reply Stripe/STPSourceCardDetails.swift:17]: Never mind, this is fine

File: Stripe/STPCard.swift (hunk 1 above)
Comment (yuki-stripe) [reply Stripe/STPCard.swift:269]: Yeah I'm not sure what this comment is about. I'm going to remove these commented lines

**Pull request merged**